index.js

```diff
@@ -786,6 +786,98 @@
   //. ```
   S.invert = def('invert', {g: [Z.Group]}, [g, g], Z.invert);
 
+  //# filter :: Filterable f => (a -> Boolean) -> f a -> f a
+  //.
+  //. Curried version of [`Z.filter`][]. Discards every element of the given
+  //. structure which does not satisfy the predicate.
+  //.
+  //. See also [`reject`](#reject).
+  //.
+  //. ```javascript
+  //. > S.filter(S.odd, [1, 2, 3])
+  //. [1, 3]
+  //.
+  //. > S.filter(S.odd, {x: 1, y: 2, z: 3})
+  //. {x: 1, z: 3}
+  //.
+  //. > S.filter(S.odd, S.Nothing)
+  //. Nothing
+  //.
+  //. > S.filter(S.odd, S.Just(0))
+  //. Nothing
+  //.
+  //. > S.filter(S.odd, S.Just(1))
+  //. Just(1)
+  //. ```
+  S.filter =
+  def('filter', {f: [Z.Filterable]}, [$.Predicate(a), f(a), f(a)], Z.filter);
+
+  //# reject :: Filterable f => (a -> Boolean) -> f a -> f a
+  //.
+  //. Curried version of [`Z.reject`][]. Discards every element of the given
+  //. structure which satisfies the predicate.
+  //.
+  //. See also [`filter`](#filter).
+  //.
+  //. ```javascript
+  //. > S.reject(S.odd, [1, 2, 3])
+  //. [2]
+  //.
+  //. > S.reject(S.odd, {x: 1, y: 2, z: 3})
+  //. {y: 2}
+  //.
+  //. > S.reject(S.odd, S.Nothing)
+  //. Nothing
+  //.
+  //. > S.reject(S.odd, S.Just(0))
+  //. Just(0)
+  //.
+  //. > S.reject(S.odd, S.Just(1))
+  //. Nothing
+  //. ```
+  S.reject =
+  def('reject', {f: [Z.Filterable]}, [$.Predicate(a), f(a), f(a)], Z.reject);
+
+  //# takeWhile :: Filterable f => (a -> Boolean) -> f a -> f a
+  //.
+  //. Curried version of [`Z.takeWhile`][]. Discards the first inner value
+  //. which does not satisfy the predicate, and all subsequent inner values.
```

Comment:
This description isn't consistent with the previous two. Perhaps replace "inner value" with "element"?

Reply:
Excellent suggestion! I love these small improvements. :)

:zap: